opslevel/resources_opslevel_team_tag.go

```diff
@@ -1,159 +1,234 @@
-// 	}
-// 	d.SetId("")
-// 	return nil
-// }
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+var _ resource.ResourceWithConfigure = &TeamTagResource{}
+
+var _ resource.ResourceWithImportState = &TeamTagResource{}
+
+type TeamTagResource struct {
+	CommonResourceClient
+}
+
+func NewTeamTagResource() resource.Resource {
+	return &TeamTagResource{}
+}
+
+type TeamTagResourceModel struct {
+	Id          types.String `tfsdk:"id"`
+	Key         types.String `tfsdk:"key"`
+	LastUpdated types.String `tfsdk:"last_updated"`
+	Team        types.String `tfsdk:"team"`
+	TeamAlias   types.String `tfsdk:"team_alias"`
+	Value       types.String `tfsdk:"value"`
+}
+
+func NewTeamTagResourceModel(teamTag opslevel.Tag, teamIdentifier string) TeamTagResourceModel {
+	teamResourceModel := TeamTagResourceModel{
+		Key:   types.StringValue(teamTag.Key),
+		Value: types.StringValue(teamTag.Value),
+		Id:    types.StringValue(string(teamTag.Id)),
+	}
+
+	// use either the team ID or alias based on what is used in the config
+	if opslevel.IsID(teamIdentifier) {
+		teamResourceModel.Team = types.StringValue(teamIdentifier)
+	} else {
+		teamResourceModel.TeamAlias = types.StringValue(teamIdentifier)
+	}
+
+	return teamResourceModel
+}
+
+func (teamTagResource *TeamTagResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_team_tag"
+}
+
+func (teamTagResource *TeamTagResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Team Tag Resource",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "The ID of this resource.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"last_updated": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+			},
+			"key": schema.StringAttribute{
+				Description: "The tag's key.",
+				Required:    true,
+				// TODO: can add validator
+			},
+			"value": schema.StringAttribute{
+				Description: "The tag's value.",
+				Required:    true,
+			},
+			"team": schema.StringAttribute{
+				Description: "The id of the team that this will be added to.",
+				Optional:    true,
+			},
+			"team_alias": schema.StringAttribute{
+				Description: "The alias of the team that this will be added to.",
+				Optional:    true,
```

Comment:
For both `team` and `team_alias` we should add this
```go
				Validators: []validator.String{
					stringvalidator.AtLeastOneOf(
						path.MatchRoot("team"),
						path.MatchRoot("team_alias"),
					),
				},
```
This will cover the `if data.Team.ValueString() == "" && data.TeamAlias.ValueString() == "" {` case down below

Reply:
And will let customers know at plan time before a create is attempted